### ISSUE TITLE
Create list of changed activity ids

### ISSUE THREAD

**Issue by notshi** [IATI/ckanext-iati#279]
Similar to https://github.com/IATI/ckanext-iati/issues/218, this is an ongoing issue where we bookmark an activity using its activity identifier and the next day, it has disappeared.

Often, these are still in the data but the publishers have just changed their id.

This makes it hard for us to keep track of things, fix issues, provide context, see if problems have been fixed, etc.

Could this also be done by listing previous activity ids within an activity? @amy-silcock

**Comment by gtkChop** [comment]
@JoshStanleyDevInit @notshi Yes, I need to have a look and see if we store activity id and fetch all changed ids from its revision history.

I keep you updated on this. Thanks

**Comment by notshi** [comment]
@jodiegardiner Similar to https://github.com/IATI/ckanext-iati/issues/324, could we have this list too?

**Comment by andylolz** [comment]
> Could this also be done by listing previous activity ids within an activity?

That’s right, so `other-identifier[@type="A3"]/@ref` is where any previous identifier(s) should live. I guess d-portal could parse that data into redirects?

**Comment by notshi** [comment]
Thanks, @andylolz and technically, yes.

However, the reality is not as a useful.
Here is a first [1000 results](http://d-portal.org/dquery?sql=select%0A%0Axson-%3E%3E%27%40ref%27%20as%20old_id%2C%20aid%20as%20new_id%2C%20count(*)%0A%0Afrom%20xson%20where%20root%3D%27%2Fiati-activities%2Fiati-activity%2Fother-identifier%27%20and%20xson-%3E%3E%27%40type%27%20%3D%20%27A3%27%0Aand%20aid%20!%3D%20xson-%3E%3E%27%40ref%27%0Agroup%20by%20xson-%3E%3E%27%40ref%27%2C%20aid%0A%0Aorder%20by%203%20desc%0A%0A%0Alimit%201000%3B%0A%0A) of that query - all of the previous activity identifiers referenced here are previous org identifiers.

Here's [the query ](http://d-portal.org/dquery/#select%0A%0Axson-%3E%3E'@ref'%20as%20old_id,%20aid%20as%20new_id,%20count(*)%0A%0Afrom%20xson%20where%20root='/iati-activities/iati-activity/other-identifier'%20and%20xson-%3E%3E'@type'%20=%20'A3'%0Aand%20aid%20!=%20xson-%3E%3E'@ref'%0Agroup%20by%20xson-%3E%3E'@ref',%20aid%0A%0Aorder%20by%203%20desc%0A%0A%0Alimit%201000;%0A%0A)if you want to tweak it.

It does get better if you [ignore that particular org id](http://d-portal.org/dquery/#select%0A%0Axson-%3E%3E'@ref'%20as%20old_id,%20aid%20as%20new_id,%20count(*)%0A%0Afrom%20xson%20where%20root='/iati-activities/iati-activity/other-identifier'%20and%20xson-%3E%3E'@type'%20=%20'A3'%0Aand%20xson-%3E%3E'@ref'%20!=%20'41140'%0Agroup%20by%20xson-%3E%3E'@ref',%20aid%0A%0Aorder%20by%203%20desc%0A%0A%0Alimit%201000;%0A%0A) but then there will be other publishers so you'll then need a list of publishers who publisher their previous org id as `other-identifier[@type="A3"]/@ref` so that you can ignore those.
